save-plugins/fix-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fix/FixPlugin.kt

```diff
@@ -18,22 +24,37 @@ import okio.Path.Companion.toPath
 @Suppress("INLINE_CLASS_CAN_BE_USED")
 class FixPlugin : Plugin {
     private val pb = ProcessBuilder()
+    private val diffGenerator = DiffRowGenerator.create()
+        .showInlineDiffs(true)
+        .mergeOriginalRevised(false)
+        .inlineDiffByWord(false)
+        .oldTag { start -> if (start) "[" else "]" }
+        .newTag { start -> if (start) "<" else ">" }
+        .build()
 
-    override fun execute(saveProperties: SaveProperties, testConfig: TestConfig) {
+    override fun execute(saveProperties: SaveProperties, testConfig: TestConfig): Sequence<TestResult> {
         val fixPluginConfig = testConfig.pluginConfigs.filterIsInstance<FixPluginConfig>().single()
-        discoverFilePairs(fixPluginConfig.testResources)
+        val files = discoverFilePairs(fixPluginConfig.testResources)
             .also {
                 logInfo("Discovered the following file pairs for comparison: $it")
             }
-            .forEach { (expected, test) ->
+        return sequence {
+            files.forEach { (expected, test) ->
                 pb.exec(fixPluginConfig.execCmd.split(" "), null)
                 val fixedLines = FileSystem.SYSTEM.readLines(
                     if (fixPluginConfig.inPlace) test else test.parent!! / fixPluginConfig.destinationFileFor(test).toPath()
                 )
                 val expectedLines = FileSystem.SYSTEM.readLines(expected)
-                // todo: check equality here
-                diff(expectedLines, fixedLines)
+                val status = diff(expectedLines, fixedLines).let { patch ->
```

Comment:
Maybe we can add some debug output about execution results?

Reply:
Yeah, that's a good idea. We can have a new field `debugInfo` in `TestResult`, and it can contain execution time, or `execCmd`'s stdout and stderr